### DISCUSSION TITLE
Is it possible to add Custom Properties to request marked with dtOrchestrator?

### DISCUSSION THREAD

**Discussion by gcsepregi** [Azure/azure-functions-durable-extension#2096] [Q&A]
We have the following use case which we implemented using durable functions: when we receive a request, we need to load records from some store and process them in parallel according to the parameters provided in the request. We are glad that AppInsights End-to-End tracing works for this job perfectly, but we need one final step to be able to utilize the already existing Performance and Failure pages.

We need to be able to filter the Orchestration on these pages by custom data that were provided in the request, like clientId, originating request-id, and so on. When we worked with WebJobs, it was possible to create custom dependencies and pass these data there as our WebJobs handled request/response pairs in sync so we were able to measure the performance.

But for durable functions, as they are offloaded to background processes, the nature of how these requests and dependencies are created limits us from using simple logs for our purposes, for which we can create custom queries. But it would be much better if we could just add the trigger custom tags to the current Activity, and those would be logged for every subsequent track operation (request, dependency, exception, etc).

Is it possible to do so?
<img width="977" alt="Screenshot 2022-02-25 at 11 30 49" src="https://user-images.githubusercontent.com/11832115/155708616-2d3771fe-3219-43f8-8d57-acf9e975732a.png">
<img width="505" alt="Screenshot 2022-02-25 at 11 30 57" src="https://user-images.githubusercontent.com/11832115/155708618-a48aefa5-2343-413b-9607-a59f4918d201.png">

**Comment by cgillum** [comment]
This is not an answer but tagging @bachuv since she's working on our improved end-to-end tracing story based on OpenTelemetry.

**Reply by gcsepregi** [reply]
.

**Reply by davidmrdavid** [reply]
Hi @gcsepregi, I noticed you pinged this thread just recently. @bachuv is out of office at the moment so they're unable to respond right now, but I'll reach out internally to see if we have any guidance on this.

**Reply by gcsepregi** [reply]
Actually, I just wrote a reply to the wrong thread, as you can see below. It would be great to hear back something. We already found another thread with some workarounds, but unfortunately, those are only for custom traces, metrics, and events, not for logs like requests or dependencies that are internal to durable functions.

I took a look at the current state of the repository if we could use Activities for the purpose of injecting tags/custom properties, but even the active activity Bag properties are not copied over in the internal initializer, so that's also a no go for us.

**Reply by davidmrdavid** [reply]
Thanks for the clarification. I'll still look to bring more eyes to this thread then

**Comment by gcsepregi** [comment]
We have been working on trying to add a custom telemetry initializer to correlate the logs with response code 0 as well as other missing parts. It correlated some of our LogInformation entries to the same operation id and worked partially for mentioned entries with response code 0 (I'm not sure where those entries come from originally, but they hold required information for end-to-end tracing which is missing from the final responses with response code 200)

Unfortunately, it broke other parts of the logging, so we removed the custom telemetry initializer. But this leaves us with issues not being able to correlate all the logs, requests, dependencies, metrics, and events into one batch of operation tree.

Are there any guidelines we're missing, or any suggestions to put the above-mentioned custom properties to system-generated request and dependency entries?
